### ISSUE TITLE
Errors when using e.shap_values() on networks with BatchNormalization OR GlobalMaxPooling

### ISSUE THREAD

**Issue by larceci89** [shap/shap#1761]
Dear all, 

I am finding two different errors when computing SHAP values on networks with a **BatchNormalization** OR a **GlobalMaxPooling** layer. To test this, you can take your notebook *"Front Page DeepExplainer MNIST Example"* at https://github.com/slundberg/shap/blob/master/notebooks/image_examples/image_classification/Front%20Page%20DeepExplainer%20MNIST%20Example.ipynb, and insert one BatchNormalization() or one GlobalMaxPooling2D() layer (substituting the Dropout(0.25) layer, for example). 
Then, errors appear when computing
`shap_values = e.shap_values(x_test[1:5])`
and they are: 

--- BATCH NORMALIZATION:
**LookupError**: gradient registry has no entry for: shap_FusedBatchNormV3
NOTE: this issue reminds of the previous issue #1110

--- GLOBALMAXPOOLING:
**ValueError**: Dimensions must be equal, but are 10 and 12 for '{{node gradient_tape/sequential/global_max_pooling2d/truediv_1}} = RealDiv[T=DT_FLOAT](gradient_tape/sequential/global_max_pooling2d/sub_1, gradient_tape/sequential/global_max_pooling2d/sub)' with input shapes: [10,64], [10,12,12,64].
NOTE: the two mismathcing dimensions (12 and 12) are precisely the dimensions that disappear after applying the GlobalMaxPooling2D layer.

Thank you for any suggestion and/or help!

Tensorflow version: 2.3.0
shap version: 0.37.0

**Comment by SaiHK** [comment]
having the same issue with google colab, this is so frustrating I run into troubles for every deep learning interpretation library I use

**Comment by JoranMichiels** [comment]
Older issue but maybe it helps someone. I have the same problem with BatchNormalization. Similar to [this comment](https://github.com/slundberg/shap/issues/1110#issuecomment-615051444) on #1110  I added the following line before using DeepExplainer:
```
shap.explainers._deep.deep_tf.op_handlers["FusedBatchNormV3"] = shap.explainers._deep.deep_tf.linearity_1d(0)
```
This is the same entry as FusedBatchNorm (without 'V3') so I assume it is correct:
https://github.com/slundberg/shap/blob/05e6cb046ff7e82fa395e80a840c96065df38c8c/shap/explainers/_deep/deep_tf.py#L726
